web/web.go

```diff
@@ -240,9 +241,11 @@ func MustBuildServerURL(pathPrefix string) string {
 	if err != nil {
 		panic(err)
 	}
-	hostname, err := os.Hostname()
-	if err != nil {
-		panic(err)
+	if hostname == "" {
+		hostname, err = os.Hostname()
+		if err != nil {
+			panic(err)
+		}
 	}
 	return fmt.Sprintf("http://%s:%s%s", hostname, port, pathPrefix)
```

Comment:
There's another `os.Hostname()` occurrence further up in the file which should actually also be replaced. It finds any scrape URLs that are actually pointing to the local host, but there we really need the hostname instead of a URL. So this might not be so workable as a single flag.